### PR TITLE
[cmake] add `OT_BORDER_AGENT_EPSKC` in CMake options

### PR DESCRIPTION
**Pull request by abtink**
Tied to `OPENTHREAD_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ENABLE`.

### CODE REVIEW

**Review by superwhd — APPROVED** [review]
LGTM 👍

File: etc/cmake/options.cmake [hunk 1]
```diff
@@ -174,6 +174,7 @@ ot_option(OT_BACKBONE_ROUTER_DUA_NDPROXYING OPENTHREAD_CONFIG_BACKBONE_ROUTER_DU
 ot_option(OT_BACKBONE_ROUTER_MULTICAST_ROUTING OPENTHREAD_CONFIG_BACKBONE_ROUTER_MULTICAST_ROUTING_ENABLE "BBR MR")
 ot_option(OT_BLE_TCAT OPENTHREAD_CONFIG_BLE_TCAT_ENABLE "Ble based thread commissioning")
 ot_option(OT_BORDER_AGENT OPENTHREAD_CONFIG_BORDER_AGENT_ENABLE "border agent")
+ot_option(OT_BORDER_AGENT_EPSK OPENTHREAD_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ENABLE "border agent ephemeral PSK")
```
Comment (superwhd) [inline etc/cmake/options.cmake:177]: A small question: when should we call it `ePSK` and when should we call it `ePSKC`?

**Review by bukepo — APPROVED** [review]

File: etc/cmake/options.cmake (hunk 1 above)
Comment (abtink) [reply etc/cmake/options.cmake:177]: Either one is fine. The "c" in `PSKc` stands for commissioner/commissioning. This is to distinguish it from a "PSKd" used by joiner devices. I believe on a Border Agent, PSK will always refer to the "commissioner" PSK.

Comment (superwhd) [reply etc/cmake/options.cmake:177]: Thanks for the explanation!

Comment (bukepo) [reply etc/cmake/options.cmake:177]: Given the specs refer to it as `ePSKc`, is it better to keep consistent with the spec?

Comment (abtink) [reply etc/cmake/options.cmake]: Changed to `EPSKC` in new push.

**Review by jwhui — APPROVED** [review]

**Pull request merged**